libcxx/test/std/depr/depr.cpp.headers/ciso646.verify.cpp

```diff
@@ -0,0 +1,18 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// <ciso646>
+
+// check that <ciso646> is removed in C++20
+// When built with modules, <ciso646> should be omitted.
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17
+// UNSUPPORTED: clang-modules-build
```

Comment:
IIUC, these diagnostics are not emitted under Clang modules? So then the clever using _LIBCPP_DEPRECATED(...)` approach is equivalent to

```
#if !__has_feature(cxx_modules) // whatever the right way of checking that is
# warning "this header is deprecated"
#endif
```

I'm not super strongly attached, but I do think that using a plain `#warning` makes this easier to understand than the clever `using` trick.

Reply:
`__has_feature(modules)` We want this for clang modules and cxx_modules is C++20 modules

Reply:
I'm afraid that currently we can't reliably detect `-fmodules`, as `__has_feature(modules)` is currently `1` in C++20 mode even when `-fno-modules` is used, see #57432.

Reply:
Why can't we use `#if __building_module(std)`?